ts/Stock/Indicators/OBV/OBVIndicator.ts

```diff
@@ -0,0 +1,244 @@
+            // For line series.
+            values = yVal as Array<number>;
+        } else {
+            // For OHLC series.
+            values = (yVal as Array<Array<number>>).map((value: Array<number>): number => value[index]);
+        }
+
+        return values;
+    }
+
+    public getTrend(
+        curentClose: number,
+        previousClose: number
+    ): number {
+        let trend: number = void 0 as any;
+
+        if (curentClose > previousClose) {
+            trend = 1; // up
+        }
+        if (curentClose === previousClose) {
+            trend = 0; // constant
+        }
+        if (curentClose < previousClose) {
+            trend = -1; // down
+        }
+
+        return trend;
+    }
+
+    public getValues<TLinkedSeries extends LineSeries>(
+        series: TLinkedSeries,
+        params: OBVParamsOptions
+    ): (IndicatorValuesObject<TLinkedSeries>|undefined) {
+        const volumeSeries = series.chart.get(params.volumeSeriesID as string),
+            xVal: Array<number> = (series.xData as any),
+            yVal: Array<number> | Array<Array<number>> = (series.yData as any),
+            OBV: Array<Array<number>> = [],
+            xData: Array<number> = [],
+            yData: Array<number> = [];
+
+        let OBVPoint: Array<number> = [],
+            i: number = 0,
+            previousOBV: number = 0,
+            curentOBV: number = 0,
+            previousClose: number = 0,
+            curentClose: number = 0,
+            volume: Array<number>,
+            closeValues: Array<number>,
+            trend: number;
+
+        // Checks if volume series exists.
+        if (volumeSeries) {
+            closeValues = this.getCloseValues(yVal);
+            volume = ((volumeSeries as Series).yData as any);
+
+            for (i; i < closeValues.length; i++) {
+                // Add first point and qet close value.
+                if (i === 0) {
+                    OBVPoint = [xVal[i], previousOBV];
+                    previousClose = closeValues[i];
+                } else {
+                    curentClose = closeValues[i];
+                    trend = this.getTrend(curentClose, previousClose);
+
+                    if (trend === 1) {
+                        curentOBV = previousOBV + volume[i];
+                    }
+                    if (trend === 0) {
+                        curentOBV = previousOBV;
+                    }
+                    if (trend === -1) {
+                        curentOBV = previousOBV - volume[i];
+                    }
+
+                    // Add point.
+                    OBVPoint = [xVal[i], curentOBV];
+
+                    // Asing currend as previous for next iteration
+                    previousOBV = curentOBV;
+                    previousClose = curentClose;
+                }
+
+                OBV.push(OBVPoint);
+                xData.push(xVal[i]);
+                yData.push(OBVPoint[1]);
+            }
+        } else {
+            return;
```

Comment:
Could you throw an error when volume series was not found? See other indicators (A/D, MFI etc)